regression_pathology_features.py

```diff
@@ -0,0 +1,147 @@
+                           drop(['img_id'], axis=1)
+    area_df =           pd.read_csv(f'{args.features_dir}/area_ratio.csv').\
+                           drop(['img_id'], axis=1)
+    elongation_df =     pd.read_csv(f'{args.features_dir}/elongation.csv').\
+                           drop(['img_id'], axis=1)
+    rec_area_ratio_df = pd.read_csv(f'{args.features_dir}/rec_area_ratio.csv').\
+                           drop(['img_id'], axis=1)
+    irrectangularity_df = 1-rec_area_ratio_df # irrectangularity = 1-(area_ratio)
+
+    # get cxrs with at least one ground-truth segmentation for any pathology
+    neg_idx = instance_df[instance_df.eq(0).all(1)].index.values
+    pos_idx = [i for i in instance_df.index.values if i not in neg_idx]
+    instance_df_pos = instance_df.iloc[pos_idx]
+    area_df_pos = area_df.iloc[pos_idx]
+    elongation_df_pos = elongation_df.iloc[pos_idx]
+    irrectangularity_df_pos = irrectangularity_df.iloc[pos_idx]
+
+    # create regression dataframe 
+    if args.evaluate_hb:
+        regression_cols = ['pred_miou', 'miou_diff', 'pred_hitrate', 'hitrate_diff']
+    else:
+        regression_cols = ['pred_miou', 'pred_hitrate']
+
+    for y in regression_cols:
+        overall_regression = pd.DataFrame()
+        coef_summary = pd.DataFrame(columns = ['lower', 'upper', 'mean',
+            'coef_pval', 'corr', 'corr_pval', 'corr_lower', 'corr_upper', 'feature'])
+        for task in sorted(LOCALIZATION_TASKS):
+            df = pd.DataFrame()
+            data = {'pred_miou': pred_miou_results[task].values,
+                    'pred_hitrate': pred_hitrate_results[task].values,
+                    'n_instance': instance_df_pos[task].values,
+                    'area_ratio':area_df_pos[task].values,
+                    'elongation': elongation_df_pos[task].values,
+                    'irrectangularity': irrectangularity_df_pos[task].values,
+                    'img_id': hb_miou_results['img_id']}
+            if args.evaluate_hb:
+                data['miou_diff'] =     hb_miou_results[task].values \
+                                            - pred_miou_results[task].values
+                data['hitrate_diff'] =  hb_hitrate_results[task].values \
+                                            - pred_hitrate_results[task].values
+            df = pd.DataFrame(data)
+            # get cxrs with ground-truth segmentation for this task
+            df = df[df.n_instance>0]
+            overall_regression = pd.concat([overall_regression, df])
+
+        features = ['n_instance', 'area_ratio', 'elongation', 'irrectangularity']
+        for feature in features:
+            overall_regression[feature] = normalize(overall_regression[feature])
+            results = run_linear_regression(overall_regression, 'Overall', y, feature)
+            coef_summary = pd.concat([coef_summary, results])
+        # TODO: is there a reason we don't include Spearman correlation here?
```

Comment:
@alexaaag is there a reason we don't include spearman correlation in the saved results csv?

Reply:
Not really. I didn't save it because we are not reporting spearman in our Table 2.